Test failure with Sphinx 7.1.2

I am getting the following test failures with Sphinx 7.1.2 & python 3.10:

```xml
<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite name="pytest" errors="0" failures="3" skipped="0" tests="4" time="0.988" timestamp="2023-08-26T11:51:46.387839" hostname="EDORAS"><testcase classname="tests.test_basic" name="test_basic" time="0.333"><failure message="assert '&lt;document so...\n        abc' == '&lt;document so...\n        abc'&#10;  - &lt;document source=&quot;&lt;src&gt;/index.rst&quot;&gt;&#10;  + &lt;document source=&quot;&lt;src&gt;/index.rst&quot; translation_progress=&quot;{'total': 0, 'translated': 0}&quot;&gt;&#10;        &lt;paragraph&gt;&#10;            abc">sphinx_doctree = &lt;sphinx_pytest.plugin.CreateDoctree object at 0x000001CD00275FD0&gt;

    def test_basic(sphinx_doctree):
        result = sphinx_doctree("abc")
&gt;       assert (
            result.pformat()
            == '&lt;document source="&lt;src&gt;/index.rst"&gt;\n    &lt;paragraph&gt;\n        abc'
        )
E       assert '&lt;document so...\n        abc' == '&lt;document so...\n        abc'
E         - &lt;document source="&lt;src&gt;/index.rst"&gt;
E         + &lt;document source="&lt;src&gt;/index.rst" translation_progress="{'total': 0, 'translated': 0}"&gt;
E               &lt;paragraph&gt;
E                   abc

tests\test_basic.py:3: AssertionError</failure></testcase><testcase classname="tests.test_basic" name="test_no_transforms" time="0.054" /><testcase classname="tests.test_basic" name="test_with_transforms" time="0.034"><failure message="assert '&lt;document so...       header' == '&lt;document so...       header'&#10;  - &lt;document source=&quot;&lt;src&gt;/index.rst&quot;&gt;&#10;  + &lt;document source=&quot;&lt;src&gt;/index.rst&quot; translation_progress=&quot;{'total': 0, 'translated': 0}&quot;&gt;&#10;        &lt;target refid=&quot;target&quot;&gt;&#10;        &lt;section ids=&quot;header target&quot; names=&quot;header target&quot;&gt;&#10;            &lt;title&gt;&#10;                header">sphinx_doctree = &lt;sphinx_pytest.plugin.CreateDoctree object at 0x000001CD01305810&gt;

    def test_with_transforms(sphinx_doctree):
        """Return the doctree, after transforms (but not post-transforms)."""
        result = sphinx_doctree(".. _target:\n\nheader\n------\n")
&gt;       assert (
            result.pformat()
            == """\
    &lt;document source="&lt;src&gt;/index.rst"&gt;
        &lt;target refid="target"&gt;
        &lt;section ids="header target" names="header target"&gt;
            &lt;title&gt;
                header
    """.rstrip()
        )
E       assert '&lt;document so...       header' == '&lt;document so...       header'
E         - &lt;document source="&lt;src&gt;/index.rst"&gt;
E         + &lt;document source="&lt;src&gt;/index.rst" translation_progress="{'total': 0, 'translated': 0}"&gt;
E               &lt;target refid="target"&gt;
E               &lt;section ids="header target" names="header target"&gt;
E                   &lt;title&gt;
E                       header

tests\test_basic.py:27: AssertionError</failure></testcase><testcase classname="tests.test_basic" name="test_html_builder" time="0.251"><failure message="assert '&lt;document so...e=&quot;preserve&quot;&gt;' == '&lt;document so...e=&quot;preserve&quot;&gt;'&#10;  - &lt;document source=&quot;&lt;src&gt;/index.rst&quot;&gt;&#10;  + &lt;document source=&quot;&lt;src&gt;/index.rst&quot; translation_progress=&quot;{'total': 0, 'translated': 0}&quot;&gt;&#10;        &lt;paragraph&gt;&#10;            abc&#10;        &lt;comment xml:space=&quot;preserve&quot;&gt;">sphinx_doctree = &lt;sphinx_pytest.plugin.CreateDoctree object at 0x000001CD01315D50&gt;

    def test_html_builder(sphinx_doctree):
        sphinx_doctree.buildername = "html"
        result = sphinx_doctree(".. only:: html\n\n   abc\n\n.. only:: latex\n\n   xyz\n")
&gt;       assert (
            result.get_resolved_pformat()
            == '&lt;document source="&lt;src&gt;/index.rst"&gt;\n    &lt;paragraph&gt;\n        abc'
            '\n    &lt;comment xml:space="preserve"&gt;'
        )
E       assert '&lt;document so...e="preserve"&gt;' == '&lt;document so...e="preserve"&gt;'
E         - &lt;document source="&lt;src&gt;/index.rst"&gt;
E         + &lt;document source="&lt;src&gt;/index.rst" translation_progress="{'total': 0, 'translated': 0}"&gt;
E               &lt;paragraph&gt;
E                   abc
E               &lt;comment xml:space="preserve"&gt;

tests\test_basic.py:42: AssertionError</failure></testcase></testsuite></testsuites>
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -41,5 +41,5 @@ repos:
       args: [--config-file=pyproject.toml]
       additional_dependencies:
       - types-docutils
-      - sphinx~=4.1
+      - sphinx~=7.0
       - pytest

--- a/src/sphinx_pytest/plugin.py
+++ b/src/sphinx_pytest/plugin.py
@@ -9,8 +9,8 @@ from typing import Any, Iterator
 from docutils import nodes
 from docutils.core import Publisher
 import pytest
+from sphinx import version_info as sphinx_version_info
 from sphinx.environment import BuildEnvironment
-from sphinx.testing.path import path
 from sphinx.testing.util import SphinxTestApp
 
 from .builders import DoctreeBuilder
@@ -90,12 +90,18 @@ class AppWrapper:
         except AttributeError:
             return Doctrees(self.env)
 
-    def pformat(self, docname: str = "index") -> str:
+    def pformat(
+        self, docname: str = "index", pop_doc_attrs=("translation_progress",)
+    ) -> str:
         """Return an indented pseudo-XML representation.
 
-        The src directory is replaced with <src>, for reproducibility.
+        By default, the src directory is replaced with <src>, for reproducibility,
+        add the ``translation_progress`` is removed (added in sphinx 7.1).
         """
-        text = self.doctrees[docname].pformat()
+        doctree = self.doctrees[docname].deepcopy()
+        for attr_name in pop_doc_attrs:
+            doctree.attributes.pop(attr_name, None)
+        text = doctree.pformat()
         return text.replace(str(self._app.srcdir) + os.sep, "<src>/").rstrip()
 
     def get_resolved_doctree(self, docname: str = "index") -> nodes.document:
@@ -109,9 +115,18 @@ class AppWrapper:
         # https://github.com/sphinx-doc/sphinx/blob/05a898ecb4ff8e654a053a1ba5131715a4514812/sphinx/environment/__init__.py#L538
         return doctree
 
-    def get_resolved_pformat(self, docname: str = "index") -> str:
-        """Return the pformat of the doctree after post-transforms."""
-        text = self.get_resolved_doctree(docname).pformat()
+    def get_resolved_pformat(
+        self, docname: str = "index", pop_doc_attrs=("translation_progress",)
+    ) -> str:
+        """Return an indented pseudo-XML representation, after post-transforms.
+
+        By default, the src directory is replaced with <src>, for reproducibility,
+        add the ``translation_progress`` is removed (added in sphinx 7.1).
+        """
+        doctree = self.get_resolved_doctree(docname)
+        for attr_name in pop_doc_attrs:
+            doctree.attributes.pop(attr_name, None)
+        text = doctree.pformat()
         return text.replace(str(self._app.srcdir) + os.sep, "<src>/").rstrip()
 
 
@@ -141,9 +156,16 @@ class CreateDoctree:
         self.srcdir.joinpath(filename).parent.mkdir(parents=True, exist_ok=True)
         self.srcdir.joinpath(filename).write_text(content, encoding="utf8")
 
+        if sphinx_version_info >= (7, 2):
+            srcdir = self.srcdir
+        else:
+            from sphinx.testing.path import path
+
+            srcdir = path(str(self.srcdir))  # type: ignore
+
         return AppWrapper(
             self._app_cls(
-                srcdir=path(str(self.srcdir)),
+                srcdir=srcdir,  # type: ignore
                 buildername=self.buildername,
                 confoverrides=self._confoverrides,
                 **kwargs,

--- a/src/sphinx_pytest/plugin.py
+++ b/src/sphinx_pytest/plugin.py
@@ -95,8 +95,12 @@ class AppWrapper:
     ) -> str:
         """Return an indented pseudo-XML representation.
 
-        By default, the src directory is replaced with <src>, for reproducibility,
-        add the ``translation_progress`` is removed (added in sphinx 7.1).
+        The src directory is replaced with <src>, for reproducibility.
+
+        :param pop_doc_attrs: Remove these attributes of the doctree node,
+            before converting to text.
+            By default, ``translation_progress`` is removed for compatibility
+            (added in sphinx 7.1).
         """
         doctree = self.doctrees[docname].deepcopy()
         for attr_name in pop_doc_attrs:
@@ -120,8 +124,12 @@ class AppWrapper:
     ) -> str:
         """Return an indented pseudo-XML representation, after post-transforms.
 
-        By default, the src directory is replaced with <src>, for reproducibility,
-        add the ``translation_progress`` is removed (added in sphinx 7.1).
+        The src directory is replaced with <src>, for reproducibility.
+
+        :param pop_doc_attrs: Remove these attributes of the doctree node,
+            before converting to text.
+            By default, ``translation_progress`` is removed for compatibility
+            (added in sphinx 7.1).
         """
         doctree = self.get_resolved_doctree(docname)
         for attr_name in pop_doc_attrs:
@@ -156,16 +164,17 @@ class CreateDoctree:
         self.srcdir.joinpath(filename).parent.mkdir(parents=True, exist_ok=True)
         self.srcdir.joinpath(filename).write_text(content, encoding="utf8")
 
+        srcdir: Any
         if sphinx_version_info >= (7, 2):
             srcdir = self.srcdir
         else:
             from sphinx.testing.path import path
 
-            srcdir = path(str(self.srcdir))  # type: ignore
+            srcdir = path(str(self.srcdir))
 
         return AppWrapper(
             self._app_cls(
-                srcdir=srcdir,  # type: ignore
+                srcdir=srcdir,
                 buildername=self.buildername,
                 confoverrides=self._confoverrides,
                 **kwargs,

